themes/website/static/js/osdetect.js

```diff
@@ -183,28 +197,67 @@ if (window.navigator.userAgent.indexOf("Windows Phone") != -1) {
 	All the magic stuff the makes the buttons change and stuff
 */
 
+// Set platform header
+document.getElementById("downloadHeader").innerHTML = "Download Tox for " + OSName;
+
 // Set platform toxlock
 document.getElementById("platImg").src = "theme/img/plat/" + OSName.toLowerCase() + ".svg";
 
 // Remove normal button
 document.getElementById("defaultButton").innerHTML = "";
 
+var buttonArea = document.getElementById("buttonArea");
+
+
 // Loop through all links and make buttons and info modals
-for (var i = 0; i < clients.length; i++) {
+var length = clients.length;
+for (var i = 0; i < length; i++) 
+{
 	var client = clients[i];
 
-	var button = "\
-	<a id='link" + i + "' href='" + client.dlLink + "' class='button large-button download'>\
-		<span class='fa fa-" + client.icon + "'>&nbsp;</span>" + client.title + "\
-	</a>";
+	if (!client.icon)
+		client.icon = "";
 
-	var infoButton = "";
+	if (client.faicon)
+		client.faicon = "fa fa-" + client.faicon;
+	else
+		client.faicon = "";
+
+	// create button
+	var button = document.createElement("A");
+	button.id = "link" + i;
+	button.href = client.dlLink;
+	button.className = "button large-button download";
+
+	// create icon
+	var buttonIcon = document.createElement("SPAN");
+	var iconClass = client.faicon;
+
+	if (!client.faicon)
+		iconClass = client.icon + " icon-distro-small";
+	
+	buttonIcon.className = iconClass;
+
+	// append icon and client name to the button
+	button.appendChild(buttonIcon);
+	button.innerHTML += " " + client.title;
+
+
+	// create info button if needed
+	var infoButton;
 	if (client.desc) {
-		infoButton = "\
-		<a class='button large-button' style='box-shadow:none;color:#368CCA;' href='clients.html#" + client.name + "' title='More info'>\
-			<span class='fa fa-info-circle'></span>\
-		</a>";
+		infoButton = document.createElement("A");
+		infoButton.className = "button large-button info-button";
+		infoButton.href = "clients.html#" + client.name;
+		infoButton.title = "More info";
+		infoButton.innerHTML = "<span class='fa fa-info-circle'></span>";
 	}
 
-	document.getElementById("buttonArea").innerHTML = document.getElementById("buttonArea").innerHTML + button + infoButton + "<br/>";
+	// append created button and info button (if created)
+	buttonArea.appendChild (button);
+	if (infoButton)
+		buttonArea.appendChild(infoButton);
+
+	buttonArea.innerHTML += "<br/>";
+
```

Comment:
Another nitpick: I don't see a lot of blank lines after the start or before the end of scopes in these files. What you've changed should probably stick to that.

Reply:
So I shouldn't separate things with new lines? I think it increases readability, but maybe it's just me.

Reply:
I don't think
```js
	buttonArea.appendChild (button);
	if (infoButton)
		buttonArea.appendChild(infoButton);

	buttonArea.innerHTML += "<br/>";

}
```
is more readable than
```js
	buttonArea.appendChild (button);
	if (infoButton)
		buttonArea.appendChild(infoButton);

	buttonArea.innerHTML += "<br/>";
}
```
The only place that I really noticed you using different whitespace from what was before is before or after braces.

Reply:
I agree that both those examples are as readable. I thought you meant something different.

Reply:
Also a good practice to put braces even for one line if. Allow faster refactoring/less issues with (bad) minifuication.

If you really wants it to be online if do this:

```js
infoButton && buttonArea.appendChild(infoButton);
```

This works because the `&&` operator always evaluate it's right instruction if the left instruction was true. Else it simply ignores the right instruction.

Reply:
Didn't know of that. Good to know!